libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/voice/NavigationInstructionPlayer.java

```diff
@@ -98,14 +94,17 @@ public void onDone() {
   }
 
   @Override
-  public void onError() {
+  public void onError(boolean isMapboxPlayer) {
     if (instructionListener != null) {
-      instructionListener.onError();
+      instructionListener.onError(isMapboxPlayer);
+    }
+
+    if (isMapboxPlayer) { // If mapbox player failed, try android speech player
```

Comment:
I'm not using the comments to explain complicated logic, just to explain the business logic so someone doesn't later look at it and wonder why we're doing that and remove it. Also oops, that's supposed to be an `else`

Reply:
My point here is that we can remove the comment fixing the naming/extracting into `private` methods and take advantage of a proper naming to express that business logic. This way comments will become superfluous so they won't be necessary. Does that make sense?
Also we need to be really careful with comments because if you change the logic but not the comment, that comment starts lying and can confuse someone who 👀 the code (including your future you 😛). Unfortunately, this happens often 😓 and that's why I'm not a big fan of code comments.